### PR TITLE
API endpoint for logging in

### PR DESCRIPTION
**Pull request by thesuss**
PT Story: 
https://www.pivotaltracker.com/story/show/132242533

Changes proposed in this pull request:
- Create API endpoint for logging in
- Route logins through a new `API::LoginController`
- Spec tests for successful and failed logins

What I have learned working on this feature: 
- how to update "regular" devise with things needed for devise token auth
- Don't use the Postman login things - put `email` and `password` in headers, för fan!

screen shot 2016-10-13 at 8 45 19 pm

Ready for review!

### CODE REVIEW

File: db/migrate/20161013135832_add_devise_auth_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class AddDeviseAuthToUsers < ActiveRecord::Migration[5.0]
+  def change
+    change_table(:users) do |t|
+      ## Required
+      t.string :provider, :null => false, :default => "email"
```
Comment (tochman) [inline db/migrate/20161013135832_add_devise_auth_to_users.rb]: Hash format. plz skip the hashrocket

Comment (thesuss) [reply db/migrate/20161013135832_add_devise_auth_to_users.rb]: We took this from cooper challenge, and didn't reflect over it.

Comment (tochman) [reply db/migrate/20161013135832_add_devise_auth_to_users.rb]: Still here...

Comment (AmberWilkie) [reply db/migrate/20161013135832_add_devise_auth_to_users.rb]: gone now

**Comment by tochman** [comment]
So I think 

```
expose: %w(access-token expiry token-type uid client format),
```

Will be needed to get the right response for setting `currentUser` in your Ionic app once you start using this endpoint.
The PR is however ready to be merged IMO.

**Comment by AmberWilkie** [comment]
I am pulling at least some info out of `currentUser` without it:
screen shot 2016-10-13 at 9 07 50 pm

Comes from: 

```
$auth.submitLogin($scope.loginData)
                .then(function (response) {
                    $ionicLoading.hide();
                    // $scope.state.go('tab-restaurants');
                    console.log('um... it worked?' + $scope.currentUser.name);
                })
```

**Comment by tochman** [comment]
Yeah.. I'm worried that the tokens will not be returned with the response. But let's cross that bridge when we get there... ;-)

**Comment by AmberWilkie** [comment]
As far as I can tell, the only thing that line does is add this in the header: `Access-Control-Expose-Headers →access-token, expiry, token-type, uid, client`
My headers look like this without it:

```
Access-Control-Allow-Credentials →true
Access-Control-Allow-Methods →GET, POST
Access-Control-Allow-Origin →chrome-extension://fhbjgbiflinjbdggehcddcbncdddomop
Access-Control-Max-Age →0
Cache-Control →max-age=0, private, must-revalidate
Content-Type →application/json; charset=utf-8
ETag →W/"7ea8544f015bb68c6d7da336b41b3cd5"
Transfer-Encoding →chunked
Vary →Origin
X-Content-Type-Options →nosniff
X-Frame-Options →SAMEORIGIN
X-Request-Id →03ed7341-91d9-4428-b0b2-163bc9fe37d7
X-Runtime →0.471545
X-XSS-Protection →1; mode=block
access-token →eZJUVnD8mPOuaL2Y4MM1XA
client →HI-JZ4DzTy7btPoaR3_3Jw
expiry →1477632557
token-type →Bearer
uid →amber@customer.com
```

**Pull request merged**